hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -459,11 +459,6 @@ private Pair<Integer, HoodieData<HoodieRecord>> initializeRecordIndexPartition()
     final HoodieMetadataFileSystemView fsView = new HoodieMetadataFileSystemView(dataMetaClient,
         dataMetaClient.getActiveTimeline(), metadata);
 
-    // MOR tables are not supported
-    if (!dataMetaClient.getTableType().equals(HoodieTableType.COPY_ON_WRITE)) {
-      throw new HoodieMetadataException("Only COW tables are supported with record index");
-    }
-
```

Comment:
Would prefer to land it in a separate commit. I guess #9017 will land earlier anyway.

Reply:
this change will be included in functional test PR (which should be merged first). i include it here for CI to pass. when merging, this diff should be auto-resolved.